api/auth.go

```diff
@@ -81,15 +97,14 @@ func WithToken(token string) Option {
 func WithExpirationTime(t int) Option {
 	return clientFunc(func(c *Client) error {
 		c.log.Debug("setting up auth", zap.Int("expiration", t))
-
 		c.auth.expiration = t
-		c.auth.expiresAt = time.Now().Add(time.Duration(t) * time.Second)
+		c.auth.expiresAt = time.Now().UTC().Add(time.Duration(t) * time.Second)
 		return nil
 	})
 }
 
 func (c *Client) TokenExpired() bool {
-	return time.Until(c.auth.expiresAt) <= 0
+	return c.auth.expiresAt.Sub(time.Now().UTC()) <= 0
```

Comment:
I'm not sure this is correct

Should be either  
```suggestion
	return c.auth.expiresAt.Sub(time.Now().UTC()) >= 0
```
or 
```suggestion
	return time.Now().UTC().Sub(c.auth.expiresAt.)) >= 0
```